### PR TITLE
Checkbox Refactor & Update Tests

### PR DESCRIPTION
**Pull request by jlp0328**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+# 2.8.3 (2020-10-07)
+
+### Breaking changes
```
Comment (mpellerin42) [inline CHANGELOG.md:3]: when you have breaking changes you should increment minor version, so here version should be 2.9.0.

File: projects/demo/src/app/demo/pages/checkbox-page/checkbox-page.component.html [hunk 2]
```diff
@@ -4,31 +4,113 @@
...
+                >Radio label</pa-checkbox
+            >
+        </div>
+        <div *ngIf="selectedTab === 'formControl'">
+            <pa-checkbox
+            [formControl]="formControl"
+                type="checkbox"
+                [id]="id"
+                [name]="'name'"
+                [value]="model"
```
Comment (julienCarret) [inline projects/demo/src/app/demo/pages/checkbox-page/checkbox-page.component.html:73]: we should not use value if we use a formControl right ? same below...

File: projects/pastanaga-angular/src/lib/controls/toggles/checkbox/checkbox-component-form-control.spec.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+@Component({
+    template: `<div>
+        <pa-checkbox
+            #paCheckbox
+            [formControl]="formControl"
+            type="checkbox"
+            [id]="id"
+            [name]="name"
+            [value]="model"
```
Comment (julienCarret) [inline projects/pastanaga-angular/src/lib/controls/toggles/checkbox/checkbox-component-form-control.spec.ts:23]: the value should be provided in formControl right ?

File: projects/pastanaga-angular/src/lib/controls/toggles/checkbox/checkbox.component.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        super.ngOnDestroy();
+    }
+
+    toggleCheckbox() {
+        if (this._type === 'radio' && !this.model) {
+            this.isChecked = true;
+            this.onChange(true);
+        } else if (this._type === 'checkbox' || !this.model) {
+            this.isChecked = !this.isChecked;
+            this.onChange(this.isChecked);
```
Comment (julienCarret) [inline projects/pastanaga-angular/src/lib/controls/toggles/checkbox/checkbox.component.ts:72]: maybe we should add a comment above this line... just to explain that onChange won't do anything when the component is managed by ControlValueAccessor (ngModel or FormControl)... i don't know...

**Review by julienCarret — APPROVED** [review]

**Pull request merged**